core/auth/src/main/java/software/amazon/awssdk/auth/token/credentials/ProfileTokenProvider.java

```diff
@@ -120,14 +99,21 @@ public SdkToken resolveToken() {
         if (loadException != null) {
             throw loadException;
         }
+
+        ProfileFile cachedOrRefreshedProfileFile = refreshProfileFile();
+        if (isNewProfileFile(cachedOrRefreshedProfileFile)) {
+            currentProfileFile = cachedOrRefreshedProfileFile;
```

Comment:
Does this need to be synchronized?

Reply:
@zoewangg? If needed, will have to make same change to `ProfileCredentialsProvider`.

Reply:
Hmm, could `resolveToken()` be invoked by different threads _at the same time_? I'd assume no.

Reply:
Yeah, not sure what the exact use case is. I'd assume a main thread would be resolving?

Reply:
Took another look and I think we should use some locking mechanism. Here is an example of how it could be done: https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/cache/CachedSupplier.java#L185

Reply:
After offline discussion, decided to use CAS